fili-core/src/test/java/com/yahoo/bard/webservice/web/endpoints/JobsEndpointResources.java

```diff
@@ -110,6 +110,24 @@ public static PreResponseStore getPreResponseStore() {
         PreResponse errorPresResponse = new PreResponse(resultSet, errorResponseContext);
         preResponseStore.save("errorPreResponse", errorPresResponse);
 
+        //Pagination test resources
+        Result result1 = new Result(new HashMap<>(), metricValues, DateTime.parse("2016-01-12T00:00:00.000Z"));
+        Result result2 = new Result(new HashMap<>(), metricValues, DateTime.parse("2016-01-13T00:00:00.000Z"));
+        Result result3 = new Result(new HashMap<>(), metricValues, DateTime.parse("2016-01-14T00:00:00.000Z"));
+
+        List<Result> results1 = new ArrayList<>();
+        results1.add(result1);
+        results1.add(result2);
+        results1.add(result3);
+        ResultSet resultSet1 = new ResultSet(results1, schema);
+
+        ResponseContext responseContext1 = new ResponseContext();
+        responseContext1.put("headers", new MultivaluedHashMap<>());
+        responseContext1.put("apiMetricColumnNames", apiMetricColumnNames);
+        responseContext1.put("requestedApiDimensionFields", new HashMap<>());
+        PreResponse preResponse1 = new PreResponse(resultSet1, responseContext1);
+        preResponseStore.save("pagination", preResponse1);
```

Comment:
Ok, I see now. We're not storing any tickets in the `ApiJobStore`, but just in the `PreResponse`. I would suggest also storing the ticket in the `ApiJobStore`. Right now, we don't verify that the ticket is in the `ApiJobStore` before getting it from the `PreResponseStore`, but that leads to some inconsistent behavior, because the user may get a 404 if their long poll times out and there is no ticket in the ApiJobStore, but get the results if the results are ready before the async timeout. A 404 really shouldn't be affected by query parameters.

So that will be changing at some point in the future, which will cause tests using the `pagination` ticket to break if we leave the tests as is.